### PR TITLE
[CALCITE-3726] Documentation for Declaring Objects For Types Defined …

### PR DESCRIPTION
**Pull request by ritesh-kapoor**
…In Schema

### CODE REVIEW

File: site/_docs/reference.md [hunk 1]
```diff
@@ -2286,6 +2286,17 @@ Note:
...
   input rows into the specified order before performing aggregation.
 
+
+#### Declaring Objects For Types Defined In Schema
+After an object type is defined and installed in the schema, you can use it to declare objects in any SQL block. For example, you can use the object type to specify the datatype of an attribute, column, variable, bind variable, record field, table element, formal parameter, or function result. At run time, instances of the object type are created; that is, objects of that type are instantiated. Each object can hold different values.
+
+e.g.:
+```SQL
+employee_typ(315, 'Francis', 'Logan', 'FLOGAN',
+        '555.777.2222', '01-MAY-04', 'SA_MAN', 11000, .15, 101, 110,
```
Comment (danny0405) [inline site/_docs/reference.md]: Also give DDL that define employee_typ and address_typ.

Comment (ritesh-kapoor) [reply site/_docs/reference.md]: Updated documentation :)

File: site/_docs/reference.md [hunk 2]
```diff
@@ -2290,7 +2290,32 @@ Note:
 #### Declaring Objects For Types Defined In Schema
 After an object type is defined and installed in the schema, you can use it to declare objects in any SQL block. For example, you can use the object type to specify the datatype of an attribute, column, variable, bind variable, record field, table element, formal parameter, or function result. At run time, instances of the object type are created; that is, objects of that type are instantiated. Each object can hold different values.
```
Comment (danny0405) [inline site/_docs/reference.md]: Why this part is under `JSON Functions`, i think we should move it below the `DDL Extensions` part as a top level.

Comment (ritesh-kapoor) [reply site/_docs/reference.md]: @danny0405 Updated, kindly check if it is the right place?

File: site/_docs/reference.md [hunk 3]
```diff
@@ -2804,6 +2805,44 @@ To enable, include `calcite-server.jar` in your class path, and add
...
+```
+
+We can declare objects of type `employee_typ` and `address_typ` :
+
+```SQL
+employee_typ(315, 'Francis', 'Logan', 'FLOGAN',
+        '555.777.2222', '01-MAY-04', 'SA_MAN', 11000, .15, 101, 110,
+         address_typ('376 Mission', 'San Francisco', 'CA', '94222'))
+```
+
```
Comment (chunweilei) [inline site/_docs/reference.md]: It seems not a valid SQL.

Comment (ritesh-kapoor) [reply site/_docs/reference.md]: This is just an example to show how new objects can be created :)

**Comment by ritesh-kapoor** [comment]
Please let me know if any changes are still pending in this PR :)

File: site/_docs/reference.md [hunk 4]
```diff
@@ -2804,6 +2805,44 @@ To enable, include `calcite-server.jar` in your class path, and add
 to the JDBC connect string (see connect string property
 [parserFactory]({{ site.apiRoot }}/org/apache/calcite/config/CalciteConnectionProperty.html#PARSER_FACTORY)).
 
+
+
+#### Declaring Objects For Types Defined In Schema
+After an object type is defined and installed in the schema, you can use it to declare objects in any SQL block. For example, you can use the object type to specify the datatype of an attribute, column, variable, bind variable, record field, table element, formal parameter, or function result. At run time, instances of the object type are created; that is, objects of that type are instantiated. Each object can hold different values.
```
Comment (danny0405) [inline site/_docs/reference.md:2811]: Can you move it under `In createFunctionStatement and usingFile, classNameLiteral and filePathLiteral are character literals.` which is the end of the whole document.

Comment (ritesh-kapoor) [reply site/_docs/reference.md]: Done:)

**Pull request closed without merging**